arch/arm/core/aarch32/thread.c

```diff
@@ -477,6 +477,14 @@ FUNC_NORETURN void z_arm_switch_to_main_no_multithreading(
 	"msr  PSPLIM, %[_psplim]\n\t"
 #endif
 	"msr  PSP, %[_psp]\n\t"       /* __set_PSP(psp) */
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
+	"cpsie i\n\t"         /* enable_irq() */
+#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+	"cpsie if\n\t"		/* __enable_irq(); __enable_fault_irq() */
+	"mov r3, #0\n\t"
+	"msr   BASEPRI, r3\n\t"	/* __set_BASEPRI(0) */
+	"isb\n\t"
+#endif
```

Comment:
`isb` should placed after the `#endif` so that it is executed also for the `CONFIG_ARMV6_M_ARMV8_M_BASELINE` case.

Reply:
Ah, good catch, thanks, will do :)

Reply:
done